evm/test/TrimmedAmount.t.sol

```diff
@@ -218,4 +220,106 @@ contract TrimmingTest is Test {
 
         assertEq(expectedIsLt, isLt);
     }
+
+    // invariant: forall (x: uint256, y: uint8, z: uint8),
+    //            (x <= type(uint64).max, y <= z)
+    //                    => (x.trim(x, 8).untrim(y) == x)
+    function testFuzz_trimIsLeftInverse(
+        uint256 amount,
+        uint8 fromDecimals,
+        uint8 toDecimals
+    ) public {
+        uint256 amt = bound(amount, 1, type(uint64).max);
+        vm.assume(fromDecimals <= 50);
+        vm.assume(toDecimals <= 50);
+
+        // NOTE: this is guaranteeed by trimming
+        vm.assume(fromDecimals <= 8 && fromDecimals <= toDecimals);
```

Comment:
Can you explain the guarantee here? I'm not finding the relevant code in the TrimmedAmount.sol file (and I don't see it on the Rust side either).

I think it is probably best to remove this check so that the fuzzer can actually guarantee that checks like these are performed.

Reply:
As written I believe the fuzzer misses this code path:

https://github.com/wormhole-foundation/example-native-token-transfers/blob/49d5ab0db6ca162e94e757e3e8aec0747ad78b9c/evm/src/libraries/TrimmedAmount.sol#L140-L142

Reply:
i don't believe it does? `toDecimals` and `fromDecimals` are flipped here: https://github.com/wormhole-foundation/example-native-token-transfers/blob/92e9f3bc5e9bc26d9d5ab463aa4777a585426401/evm/test/TrimmedAmount.t.sol#L244

Reply:
What's the reason for flipping them? The `trim` function should take `fromDecimals` as the first parameter.

Also if they're flipped, then isn't the code missing the inverse branch?

Generally I'm not clear why the two decimal amounts are being constrained like this

Reply:
it still goes through the other code path on untrimming but can simplify this without restricting the input